Model/Order/HydrateOrderFromQuote.php

```diff
@@ -218,6 +218,7 @@ private function formatCartItems(array $cartItems): array
             $cartItem = $this->productFactory->create()->load($item['id']);
             $item['sku'] = $cartItem->getSku();
             $item['vendor'] = '';
+            $item['weight'] = (int)ceil($item['weight']);
```

Comment:
Out of curiosity, what will happen on items that don't have a weight setup, or weight is empty as Magento allow it to be that way?

Reply:
Our existing hydration logic defaults the weight to a value of 0 if it is not already set.